--- dropbear-2014.63/Makefile.in.0000~	2014-02-19 14:05:24.000000000 +0000
+++ dropbear-2014.63/Makefile.in	2014-05-19 19:11:46.573402023 +0000
@@ -13,14 +13,21 @@ ifndef PROGRAMS
 	PROGRAMS=dropbear dbclient dropbearkey dropbearconvert
 endif
 
+ifdef WHOLE_PROGRAM
+include libtomcrypt/Makefile
+STATIC_LTC := $(patsubst %,$(srcdir)/libtomcrypt/%,$(subst .o,.c,$(OBJECTS)))
+include libtommath/Makefile
+STATIC_LTM := $(patsubst %,$(srcdir)/libtommath/%,$(subst .o,.c,$(OBJECTS)))
+else
 STATIC_LTC=libtomcrypt/libtomcrypt.a
 STATIC_LTM=libtommath/libtommath.a
+endif
 
 LIBTOM_LIBS=@LIBTOM_LIBS@
 
 ifeq (@BUNDLED_LIBTOM@, 1)
 LIBTOM_DEPS=$(STATIC_LTC) $(STATIC_LTM) 
-CFLAGS+=-I$(srcdir)/libtomcrypt/src/headers/
+CFLAGS+=-I$(srcdir)/libtommath/ -I$(srcdir)/libtomcrypt/src/headers/
 LIBTOM_LIBS=$(STATIC_LTC) $(STATIC_LTM) 
 endif
 
@@ -180,8 +187,18 @@ scp: $(SCPOBJS)  $(HEADERS) Makefile
 MULTIOBJS=
 ifeq ($(MULTI),1)
 	MULTIOBJS=dbmulti.o $(sort $(foreach prog, $(PROGRAMS), $($(prog)objs)))
+ifdef WHOLE_PROGRAM
+	MULTIOBJS :=$(subst .o,.c,$(MULTIOBJS)) $(LIBTOM_LIBS)
 	CFLAGS+=$(addprefix -DDBMULTI_, $(PROGRAMS)) -DDROPBEAR_MULTI
 endif
+endif
+
+ifdef WHOLE_PROGRAM
+aes_enc: 
+	make -C libtomcrypt src/ciphers/aes/aes_enc.c
+dropbearmulti: $(HEADERS) aes_enc Makefile options.h
+	$(CC) $(CFLAGS) $(LDFLAGS) -o dropbearmulti$(EXEEXT) $(MULTIOBJS) $(LIBS) @CRYPTLIB@
+else
 
 dropbearmulti$(EXEEXT): $(HEADERS) $(MULTIOBJS) $(LIBTOM_DEPS) Makefile
 	$(CC) $(LDFLAGS) -o $@ $(MULTIOBJS) $(LIBTOM_LIBS) $(LIBS) @CRYPTLIB@
@@ -189,6 +206,7 @@ dropbearmulti$(EXEEXT): $(HEADERS) $(MUL
 multibinary: dropbearmulti$(EXEEXT)
 
 multilink: multibinary $(addprefix link, $(PROGRAMS))
+endif
 
 link%:
 	-rm -f $*$(EXEEXT)
--- dropbear-2014.63/includes.h.0000~	2014-05-19 19:17:34.916544691 +0000
+++ dropbear-2014.63/includes.h	2014-05-19 19:16:25.142328102 +0000
@@ -125,11 +125,11 @@
 #endif
 
 #ifdef BUNDLED_LIBTOM
-#include "libtomcrypt/src/headers/tomcrypt.h"
 #include "libtommath/tommath.h"
+#include "libtomcrypt/src/headers/tomcrypt.h"
 #else
-#include <tomcrypt.h>
 #include <tommath.h>
+#include <tomcrypt.h>
 #endif
 
 
--- dropbear-2014.63/libtomcrypt/Makefile.in.0000~	2014-02-19 14:05:24.000000000 +0000
+++ dropbear-2014.63/libtomcrypt/Makefile.in	2014-05-19 19:11:46.574402070 +0000
@@ -19,7 +19,7 @@ srcdir=@srcdir@
 
 # Compilation flags. Note the += does not write over the user's CFLAGS!
 # The rest of the flags come from the parent Dropbear makefile
-CFLAGS += -c -I$(srcdir)/src/headers/ -I$(srcdir)/../ -DLTC_SOURCE -I$(srcdir)/../libtommath/
+CFLAGS += -I$(srcdir)/src/headers/ -I$(srcdir)/../ -DLTC_SOURCE -I$(srcdir)/../libtommath/
 
 # additional warnings (newer GCC 3.4 and higher)
 ifdef GCC_34
@@ -232,6 +232,10 @@ COMPRESSED=crypt-$(VERSION).tar.bz2 cryp
 default:library
 
 #ciphers come in two flavours... enc+dec and enc 
+src/ciphers/aes/aes_enc.c: src/ciphers/aes/aes.c
+	echo '#define ENCRYPT_ONLY' >> $@
+	cat $< >> $@ 
+
 src/ciphers/aes/aes_enc.o: src/ciphers/aes/aes.c src/ciphers/aes/aes_tab.c
 	$(CC) $(CFLAGS) -DENCRYPT_ONLY -c $< -o src/ciphers/aes/aes_enc.o
 
